src/snowflake/connector/result_batch.py

```diff
@@ -438,8 +440,10 @@ def _load(self, response: "Response") -> List:
 
     def _parse(
         self, downloaded_data
-    ) -> Union[Iterator[Union[Dict, Exception]], Iterator[Union[Tuple, Exception]]]:
+    ) -> Union[List[Union[Dict, Exception]], List[Union[Tuple, Exception]]]:
         """Parses downloaded data into its final form."""
+        logger.debug(f"parsing for result batch of size {self.rowcount}")
```

Comment:
This is a good idea - I hadn't considered extracting data from the url but this will be useful to identify the batches in the logs. I'll update all the logs.